src/UnigramWordPredictor.java

```diff
@@ -49,9 +50,25 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    * @param scanner the Scanner to read the training text from
    */
   public void train(Scanner scanner) {
+    neighborMap = new HashMap<String, List<String>>(); // instantiate neighborMap as HashMap
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    for (int i = 0; i < trainingWords.size(); i++) {
+      String current = trainingWords.get(i); // save current token in arraylist
+      String next = null;
+
+      if (i < trainingWords.size()-1) {
+        next = trainingWords.get(i+1); // save follow-up token in arraylist
+      }
+
+      if (!neighborMap.containsKey(current)) {
+        neighborMap.put(current, new ArrayList<>()); // put current token as key, create arraylist tied to it for follow-up words
+      }
```

Comment:
Nice! As an alternative, can you think about how you could write this using `putIfAbsent` or `computeIfAbsent` instead?